pyterrier/apply_base.py

```diff
@@ -160,6 +194,16 @@ def __init__(self, fn,  *args, **kwargs):
 
     def __repr__(self):
         return "pt.apply.doc_features()"
+    
+    def transform_iter(self, iterdict):
+        fn = self.fn
+        # we assume that the function can take a dictionary as well as a pandas.Series. As long as [""] notation is used
+        # to access fields, both should work
+        def gen():
```

Comment:
Why does this wrap an inner function instead of making gen the function itself? (same question for the ones below too)